src/GafferImageUI/ImageView.cpp

```diff
@@ -480,6 +478,19 @@ class Box2iGadget : public GafferUI::Gadget
 				return;
 			}
 
+			V2f planarScale = viewportPlanarScale();
+			const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
+			const ImageGadget *imageGadget = static_cast<const ImageGadget *>( viewportGadget->getPrimaryChild() );
+			// Weird hack to reverse engineer imageGadget->format().getPixelAspect() since it's not public
```

Comment:
I think it'd be OK to make this public. Alternatively it can be retrieved from `imageGadget->getImage()->format()`. However we retrieve it though, we need to consider that we're triggering a computation and exceptions might be thrown (I tested with an ImageReader with an invalid filename, and got reams of unwanted error output in the shell).

Reply:
It seems worthwhile to use the imageGadget's cache, rather the recomputing the plug, and I didn't want to mess around with making it public right now ( if format() is public, then dataWindow(), and maybe channels() should probably be public too ).  For the moment, I've kept using pixelAt, but I've added a catch.  In the long run, it would probably be cleaner if ImageGadget->format() didn't throw and was public.